sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedReadFromUnboundedSource.java

```diff
@@ -83,10 +83,18 @@
     this.maxReadTime = maxReadTime;
   }
 
+  /**
+   * Returns an adapted {@link BoundedSource} wrapping the underlying {@link UnboundedSource},
+   * with the specified bounds on number of records and read time.
+   */
+  public BoundedSource<ValueWithRecordId<T>> getAdaptedSource() {
```

Comment:
This is a new public method that is fundamental enough that I could see users coming to depend on it. I approve of it from the position that a `PTransform` subclass should feel free to expose its definitive pieces, much like `ParDo` exposes the `DoFn`.

Can you mark it `@Experimental`? We need to start increasing our discipline about being very careful not to expose public APIs that might be considered stable too early.